### ISSUE TITLE
BUG: test failures in NetworkX related to `sparse.linalg.splu`

### ISSUE THREAD

**Issue by rossbar** [scipy/scipy#18752]
### Describe your issue.

There are 33 new test failures in the NetworkX test suite with the release of scipy 1.11.0. I haven't gone through in detail yet, but most seem to be related to linalg - specifically the LU solver.

Possibly related to #18751

Not sure why this wasn't caught by NX's tests against the pre-release/dev version of scipy, but that's a separate issue!

### Reproducing Code Example

```python
>>> import networkx as nx
>>> G = nx.complete_graph(4)
>>> L = nx.laplacian_matrix(G)
>>> L.toarray()
array([[ 3, -1, -1, -1],
       [-1,  3, -1, -1],
       [-1, -1,  3, -1],
       [-1, -1, -1,  3]])
>>> import scipy as sp
>>> lusolve = sp.sparse.linalg.factorized(L.tocsc())
Traceback (most recent call last)
   ...
TypeError: rowind and colptr must be of type cint
```

### Error message

```shell
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[9], line 1
----> 1 lusolve = sp.sparse.linalg.factorized(L.tocsc())

File ~/.virtualenvs/nx-dev/lib/python3.11/site-packages/scipy/sparse/linalg/_dsolve/linsolve.py:576, in factorized(A)
    574     return solve
    575 else:
--> 576     return splu(A).solve

File ~/.virtualenvs/nx-dev/lib/python3.11/site-packages/scipy/sparse/linalg/_dsolve/linsolve.py:414, in splu(A, permc_spec, diag_pivot_thresh, relax, panel_size, options)
    411 if (_options["ColPerm"] == "NATURAL"):
    412     _options["SymmetricMode"] = True
--> 414 return _superlu.gstrf(N, A.nnz, A.data, A.indices, A.indptr,
    415                       csc_construct_func=csc_construct_func,
    416                       ilu=False, options=_options)

TypeError: rowind and colptr must be of type cint
```

### SciPy/NumPy/Python version and system information

```shell
1.11.0 1.25.0 sys.version_info(major=3, minor=11, micro=3, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/local/include
    lib directory: /usr/local/lib
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP= HASWELL MAX_THREADS=2
    pc file directory: /usr/local/lib/pkgconfig
    version: 0.3.21.dev
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/local/include
    lib directory: /usr/local/lib
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP= HASWELL MAX_THREADS=2
    pc file directory: /usr/local/lib/pkgconfig
    version: 0.3.21.dev
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.10.4
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 0.29.35
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  pythran:
    include directory: ../../tmp/pip-build-env-gltvk10g/overlay/lib/python3.11/site-packages/pythran
    version: 0.13.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /opt/python/cp311-cp311/bin/python
  version: '3.11'
```

**Comment by dschult** [comment]
Looks like the NetworkX tests are failing with SciPy v1.11.0 due to SuperLU requiring a downcast of the index type to intc. That is (I think) fixed in [scipy-#18644](https://github.com/scipy/scipy/pull/18644)  but it didn't get into the release... :{

**Comment by rossbar** [comment]
I tried #18644 on @dschult 's suggestion and it does indeed resolve this issue (though #18644 required a couple tweaks to get it running - see https://github.com/scipy/scipy/pull/18644#discussion_r1242337966).
